vms/avm/states/state.go

```diff
@@ -576,3 +578,50 @@ func (s *state) writeStatuses() error {
 	}
 	return nil
 }
+
+func (s *state) rejectedTxs() ([]ids.ID, error) {
+	var txIDs []ids.ID
+	iter := s.statusDB.NewIterator()
+	for iter.Next() {
+		key := iter.Key()
+		val := iter.Value()
+		valStatus, err := database.ParseUInt32(val)
+		if err != nil {
+			return []ids.ID{}, err
+		}
+		if choices.Status(valStatus) == choices.Rejected {
+			txID, err := ids.ToID(key)
+			if err != nil {
+				return []ids.ID{}, err
+			}
+			txIDs = append(txIDs, txID)
+			// remove txID from dbStatus
+			if err := database.ClearPrefix(s.statusDB, s.statusDB, key); err != nil {
+				return []ids.ID{}, err
+			}
```

Comment:
Why are we using `database.ClearPrefix`? This is just removing a single key. `s.statusDB.Delete(key)`.

Reply:
updated.